OMCompiler/Compiler/BackEnd/IndexReduction.mo

```diff
@@ -1387,6 +1389,44 @@ algorithm
   end match;
 end countStateCandidates;
 
+protected function countStateCandidatesWithNever
+"author: kabdelhak FHB 2019-07
+  count the number of states with stateSelect.never"
+  input BackendDAE.Var inVar;
+  input Integer inCount;
+  output BackendDAE.Var outVar;
+  output Integer outCount;
+algorithm
+  (outVar,outCount) := match inVar
+    local
+      Integer diffcount,statecount;
+      Boolean b;
+    case BackendDAE.VAR(varKind=BackendDAE.STATE(index=1))
+      equation
+        // do only count states with stateSelect.never
+        b = varStateSelectNever(inVar);
+        statecount = if b then inCount+1 else inCount;
+      then (inVar, statecount);
+
+    case BackendDAE.VAR(varKind=BackendDAE.STATE(derName=SOME(_)))
+      equation
+        // do only count states with stateSelect.never, but ignore only higest state
+        b = varStateSelectNever(inVar);
+        statecount = if not b then inCount+1 else inCount;
+      then (inVar, statecount);
+
+    case BackendDAE.VAR(varKind=BackendDAE.STATE(index=diffcount,derName=NONE()))
+      equation
+        statecount = diffcount + inCount;
+        // do only count states with stateSelect.never, but ignore only higest state
```

Comment:
Copy pasted comment?

Reply:
> Copy pasted comment?

I changed it up a bit so that it resembles what it should do. 
Actually i don't fully understand the original function and why the index is used here, but as i said the function is not in use jet.
Do you see a problem with the description?

Reply:
Nah! I was just waiting for my own stuff to compile.. I saw the same comment in two places that's why I wrote. It gave the impression that both bodies did the same thing which they did not quite do